### PR TITLE
fix(deps): Migrate MFA code to our codebase and bump djangorestframework

### PR DESCRIPTION
**Pull request by gagantrivedi**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [ ] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

- Because Trench is no longer maintained, this commit adds all the code that we depend on.
- Bump djangorestframework
- Fixes update feature external resource endpoint

## How did you test this code?

The following scenarios were tested manually:

  -  Generate MFA code from the old library and log in using the code from this PR and vice versa.
   - Log in using the backup codes generated by the old library with the code from this PR and vice versa.

### CODE REVIEW

File: api/custom_auth/mfa/trench/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+__version__ = "0.3.1"
```
Comment (matthewelwell) [inline api/custom_auth/mfa/trench/__init__.py]: Do we need this? Feels like another version number that will inevitably become stale.

File: api/custom_auth/urls.py [hunk 2]
```diff
@@ -25,12 +26,13 @@
...
+    path("logout/", TokenDestroyView.as_view(), name="authtoken-logout"),
     path("", include(ffadmin_user_router.urls)),
     path("token/", delete_token, name="delete-token"),
     # NOTE: endpoints provided by `djoser.urls`
     # are deprecated and will be removed in the next Major release
     path("", include("djoser.urls")),
-    path("", include("trench.urls")),  # MFA
-    path("", include("trench.urls.djoser")),  # override necessary urls for MFA auth
+    path("", include("custom_auth.mfa.trench.urls")),  # MFA
+    # path("", include("trench.urls")),  # MFA
```
Comment (matthewelwell) [inline api/custom_auth/urls.py]: Is there a reason that this commented out? If not, let's remove it. If there is a reason for this to still be here, let's add a comment explaining why it's commented out.

**Review by matthewelwell — APPROVED** [review]
Approving with 2 minor comments.

File: api/custom_auth/mfa/trench/__init__.py (hunk 1 above)
Comment (gagantrivedi) [reply api/custom_auth/mfa/trench/__init__.py]: Yeah, we don't need this

File: api/custom_auth/urls.py (hunk 2 above)
Comment (gagantrivedi) [reply api/custom_auth/urls.py]: Good catch! I forgot to remove this after our discussion around removing endpoints that are not used by UI

**Review by matthewelwell — APPROVED** [review]

**Comment by matthewelwell** [comment]
@gagantrivedi this looks good to me now, I think we can resolve the merge conflicts and merge it in.

**Pull request merged**